File: tests/fixtures/taint_string_ops.js
function templateLiteralPropagation(req, res) {
    const userInput = req.query.name;
    const html = `<h1>${userInput}</h1>`;
    res.send(html);
}

function concatPropagation(req, res) {
    const userInput = req.query.name;
    const html = "<h1>" + userInput + "</h1>";
    res.send(html);
}

File: tests/fixtures/taint_string_ops.py
# Line numbers are critical for tests — do not add/remove lines


def fstring_propagation(request):
    user_input = request.args.get("q")
    query = f"SELECT * FROM users WHERE name = '{user_input}'"
    cursor.execute(query)


def concat_propagation(request):
    user_input = request.args.get("q")
    query = "SELECT * FROM users WHERE name = '" + user_input + "'"
    cursor.execute(query)


def format_propagation(request):
    user_input = request.args.get("q")
    query = "SELECT * FROM users WHERE name = '{}'".format(user_input)
    cursor.execute(query)
